code/implementation/docker/common/src/main/java/io/cattle/platform/docker/util/DockerUtils.java

```diff
@@ -33,4 +35,18 @@ public static Host getHostFromContainer(ObjectManager objectManager, Instance in
         return getHostFromContainer(objectManager, instance, DockerHostConstants.KIND_DOCKER);
     }
 
+    /**
+     * The docker API allows for identification of a container by name or id.
+     * This method will return the docker container id (which is stored in the
+     * externalId field of the instance) if it is available. If it isn't
+     * available, the instance uuid will be returned, since that is currently
+     * used as the name for docker containers that do not have the docker id
+     * set.
+     * 
```

Comment:
Yes it does. I think you could maybe explain why there would be a uuid but not an id. But i don't it matters to much.